### ISSUE TITLE
Searchbox role cannot be changed and creates accessibility issue when used with CommandBar

### ISSUE THREAD

**Issue by sysofwan** [microsoft/fluentui#18271]
### Environment Information

- **Package version(s)**: 7.170.1

### Describe the issue:

It is impossible to override role="searchbox" to something more appropriate. Specifically, in a CommandBar, it is better that role="menuitem" so that screen readers announce number of items correctly.

This is partially fixed with issue #14914. However, since the external div is not focusable, screen reader will not announce number of elements when focused on the search box. The ideal fix is to make role="searchbox" overridable.

This is an issue if focus first lands on the SearchBox. There is no way for users to figure out that focus is in a CommandBar and there are other controls to the left/right of the SearchBox.

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/sysofwan/pen/MWpJrEM

#### Actual behavior:

When navigating command bar, screen reader announces:
New  subMenu  1 of 5
Upload  2 of 5
Share  3 of 5
**Search  edit  blank**
Download  5 of 5

#### Expected behavior:

When navigating command bar, screen reader should announce:
New  subMenu  1 of 5
Upload  2 of 5
Share  3 of 5
**Search  editable  4 of 5  blank**
Download  5 of 5

### Documentation describing expected behavior

Screen reader correctly announce that it is focused on the 4th item, and there are other items in CommandBar.

**Comment by gouttierre** [comment]
@sysofwan - thanks for filing this issue. and providing the codepen and version details.

@miroslavstastny - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley

**Comment by sysofwan** [comment]
This is not a regression. It's a feature request to enable overriding the role attribute, so the control can work in different context (CommandBar in this case).

**Comment by smhigley** [comment]
@sysofwan overriding the role of `searchbox` would be much, much worse than the current experience. A `menuitem` cannot be typed into, and a textbox is prohibited from having the role of `menuitem` (for good reason, it would completely trash the screen reader experience).

Our current role structure is far from ideal, and actually fairly broken in the sense that a menuitem cannot contain additional interactive items, but at least it's broken in a way that minimizes side effects.

Essentially, the issue is that menus cannot contain text inputs, and yet ours allows you to include them 😅. The real fix would be to not have a searchbox within a menu widget at all, which would require some sort of way to tell CommandBar to render as a series of grouped buttons & controls rather than an ARIA menu pattern.

I'm not sure how much we can actually do in v8, but this would be good to include in our vNext discussions. cc/ @ling1726 for vNext menu :)

Also in the meantime, I cannot emphasize this enough -- *do not override the role of the search input to be a menuitem* 😄

**Comment by sysofwan** [comment]
@smhigley thanks for clarifying. I agree, there is no guidelines for editable text fields in the wai-aria spec for menubars.

However, looks like both NVDA and Narrator is handling it pretty well. For example,  NVDA will read "Search editable 4 of 5 blank" in this situation, which is better than "Search edit blank". You can test this by manually changing role="search" -> role="menuitem" in [this codepen](https://codepen.io/sysofwan/pen/ZEKYEWP)

Can you clarify how the screen reader is broken in this case?
